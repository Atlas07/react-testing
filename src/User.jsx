import React from 'react';
import { shape, string, number } from 'prop-types';

const User = ({ data }) => (
  <div>
    <p>name - {data.name}</p>
    <p>login - {data.login}</p>
    <p>bio - {data.bio}</p>
    <p>public repos - {data.public_repos}</p>
  </div>
);

User.defaultProps = {
  bio: null,
  public_repos: null,
};

User.propTypes = {
  data: shape({
    name: string.isRequired,
    login: string.isRequired,
    bio: string,
    public_repos: number,
  }).isRequired,
};

export default User;