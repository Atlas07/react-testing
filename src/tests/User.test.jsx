import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import User from '../User';

it('renders without crashing', () => {
  const data = {
    name: 'Vitalii Shapoval',
    login: 'Atlas07',
    bio: 'Software engineer',
  }
  shallow(<User data={data} />);
});

it('renders correctly user component', () => {
  const props = {
    data: {
      name: 'Vitalii Shapoval',
      login: 'Atlas07',
      bio: 'Software engineer',
    }
  };
  const UserComponent = renderer.create(<User {...props} />).toJSON();

  expect(UserComponent).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const data = {
    name: 'Vitalii Shapoval',
    login: 'Atlas07',
    bio: 'Engineer',
    public_repos: 0,
  };
  const wrapper = mount(<User data={data} />);

  const { name, login, bio, public_repos } = wrapper.prop('data');

  expect(name).toEqual('Vitalii Shapoval');
  expect(login).toEqual('Atlas07');
  expect(bio).toEqual('Engineer');
  expect(public_repos).toEqual(0);
});

it('renders correctly only with required props', () => {
  const data = {
    name: 'Vitalii Shapoval',
    login: 'Atlas07',
  };

  const wrapper = mount(<User data={data} />);
  const { name, login } = wrapper.prop('data');

  expect(name).toEqual('Vitalii Shapoval');
  expect(login).toEqual('Atlas07');
});
