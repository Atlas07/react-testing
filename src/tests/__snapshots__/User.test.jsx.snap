// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly user component 1`] = `
<div>
  <p>
    name - 
    Vitalii Shapoval
  </p>
  <p>
    login - 
    Atlas07
  </p>
  <p>
    bio - 
    Software engineer
  </p>
  <p>
    public repos - 
  </p>
</div>
`;
